presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java

```diff
@@ -98,24 +101,57 @@ public ConnectorSplitSource getSplits(
         HudiTableLayoutHandle layout = (HudiTableLayoutHandle) layoutHandle;
         HudiTableHandle table = layout.getTable();
 
+        // Load Hudi metadata
+        ExtendedFileSystem fs = getFileSystem(session, table);
+        boolean hudiMetadataTableEnabled = isHudiMetadataTableEnabled(session);
+        HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder().enable(hudiMetadataTableEnabled).build();
+        Configuration conf = fs.getConf();
+        HoodieTableMetaClient metaClient = HoodieTableMetaClient
```

Comment:
would n't we otherwise create a metaClient here anyways? could we reuse instead of. creating a new one for data skipping alone?